ProductsManegements/Inventory.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+
+namespace SimpleInventoryManagementSystem.ProductsManegements
+{
+    internal class Inventory
+    {
+        internal static List<Product> products =new();
+
+        public void AddProduct()
+        {
+            string name = " ";
+            decimal price = 0;
+            int quantity = 0;
+            Product? product=null;
+
+            Console.ForegroundColor = ConsoleColor.Yellow;
+            Console.WriteLine("Enter the name of the product: ");
+            Console.ResetColor();
+            name = Console.ReadLine() ?? string.Empty;
+            do
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("Enter the price of the product: ");
+                Console.ResetColor();
+
+            } while (!decimal.TryParse(Console.ReadLine(), out price));
+            do
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("Enter the quantity of the product: ");
+                Console.ResetColor();
+
+            } while (!int.TryParse(Console.ReadLine(), out quantity));
+
+            product=new Product(name, price, quantity);
+            products.Add(product);
+
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine($"A new product {product.Name} is added successfully! ");
+            Console.ResetColor();
+        }
+
+        public void ViewAllProducts()
+        {
+           
+            foreach(Product product in products)
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("-----------------------");
+                Console.ResetColor();
+                Console.WriteLine(product.ShowProductDetails());
+       
+            }
+
+
+        }
+
+        public Product? SearchProduct()
```

Comment:
I would say that returning null if we didn't find the product might not be the best choice. 
search how can we do it better.